### PR TITLE
[SkyServe] New method to install cloud dependencies

### PR DESCRIPTION
**Pull request by cblmemo**
Manually install the dependencies for the cloud sdk’s on Sky Serve controller.

Tested (run the relevant ones):

- [x] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [x] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
     - [x] All Sky Serve smoke tests
     - [x] All Spot smoke tests
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/utils/controller_utils.py [hunk 1]
```diff
@@ -122,6 +123,42 @@ def from_name(cls, name: Optional[str]) -> Optional['Controllers']:
...
+        'boto3>=1.26.1 > /dev/null 2>&1',
+        # gcp
+        'pip list | grep google-api-python-client > /dev/null 2>&1 || '
+        'google-api-python-client>=2.69.0 google-cloud-storage '
+        '> /dev/null 2>&1',
+        f'{gcp.GOOGLE_SDK_INSTALLATION_COMMAND}',
+        # ibm
+        'pip list | grep ibm-cloud-sdk-core > /dev/null 2>&1 || '
+        'pip install ibm-cloud-sdk-core ibm-vpc ibm-platform-services '
+        'ibm-cos-sdk > /dev/null 2>&1',
```
Comment (Michaelvll) [inline sky/utils/controller_utils.py:144]: Seems IBM does not support spot instances?

File: sky/utils/controller_utils.py [hunk 2]
```diff
@@ -122,6 +123,42 @@ def from_name(cls, name: Optional[str]) -> Optional['Controllers']:
...
+        # oci
+        'pip list | grep oci > /dev/null 2>&1 || '
+        'pip install oci > /dev/null 2>&1'
+        # kubernetes
+        'pip list | grep kubernetes > /dev/null 2>&1 || '
+        'pip install kubernetes > /dev/null 2>&1',
```
Comment (Michaelvll) [inline sky/utils/controller_utils.py:150]: K8s also does not support spot instances? Should we move these dependencies to L155?

**Review by Michaelvll — COMMENTED** [review]
Thanks for fixing this @cblmemo! Can we mention the tests in the PR description?

**Comment by cblmemo** [comment]
@Michaelvll All smoke tests related to spot/sky serve passed. PTAL 🫡

**Review by Michaelvll — APPROVED** [review]
Thanks for the fix @cblmemo! LGTM with a comment.

**Pull request merged**